contracts/collections/sg721-base/src/contract.rs

```diff
@@ -225,27 +223,63 @@ where
         // convert from response to royalty info for storage
         let current_royalty_info = collection
             .royalty_info
-            .as_ref()
-            .map(|royalty_info| royalty_info.to_response());
+            .map(|royalty_info| royalty_info.into());
 
         let new_royalty_info = collection_msg
             .royalty_info
             .unwrap_or_else(|| current_royalty_info.clone());
 
+        // get the factory from the minter to get
+        // max_royalty, max_royalty_increase_rate
+        let minter_address = self.parent.minter(deps.as_ref())?.minter.unwrap();
+        let minter_config: MinterConfig<T> = deps
+            .querier
+            .query_wasm_smart(minter_address, &MinterQueryMsg::Config {})?;
+        let factory_address = minter_config.factory;
+        let factory_params: ParamsResponse<T> = deps
+            .querier
+            .query_wasm_smart(factory_address, &Sg2QueryMsg::Params {})?;
+        let max_royalty = factory_params.params.max_royalty_bps.bps_to_decimal();
+        let max_royalty_increase_rate = factory_params
+            .params
+            .max_royalty_increase_rate_bps
+            .bps_to_decimal();
```

Comment:
So this means we will have to migrate the factory, have we verified this won't break previous minters?

Reply:
need to migrate code ids for previous minters too